apps/ui-library/registry/default/examples/infinite-query-hook-demo.tsx

```diff
@@ -0,0 +1,147 @@
+export function InfiniteList<TableName extends SupabaseTableName>({
+  tableName,
+  columns = '*',
+  pageSize = 20,
+  trailingQuery,
+  renderItem,
+  className,
+  renderNoResults = DefaultNoResults,
+  renderEndMessage = DefaultEndMessage,
+  renderSkeleton = defaultSkeleton,
+}: InfiniteListProps<TableName>) {
+  const { data, isFetching, hasMore, fetchNextPage, isSuccess } = useInfiniteQuery({
+    tableName,
+    columns,
+    pageSize,
+    trailingQuery,
+  })
+
+  // Ref for the scrolling container
+  const scrollContainerRef = React.useRef<HTMLDivElement>(null)
+
+  // Intersection observer logic - target the last rendered *item* or a dedicated sentinel
+  const loadMoreSentinelRef = React.useRef<HTMLDivElement>(null)
+  const observer = React.useRef<IntersectionObserver | null>(null)
+
+  React.useEffect(() => {
+    if (observer.current) observer.current.disconnect()
+
+    observer.current = new IntersectionObserver(
+      (entries) => {
+        if (entries[0].isIntersecting && hasMore && !isFetching) {
+          fetchNextPage()
+        }
+      },
+      {
+        root: scrollContainerRef.current, // Use the scroll container for scroll detection
+        threshold: 0.1, // Trigger when 10% of the target is visible
+        rootMargin: '0px 0px 100px 0px', // Trigger loading a bit before reaching the end
+      }
+    )
+
+    if (loadMoreSentinelRef.current) {
+      observer.current.observe(loadMoreSentinelRef.current)
+    }
+
+    return () => {
+      if (observer.current) observer.current.disconnect()
+    }
+  }, [isFetching, hasMore, fetchNextPage])
+
+  return (
+    <div ref={scrollContainerRef} className={cn('relative h-full overflow-auto', className)}>
+      <div>
+        {isSuccess && data.length === 0 && renderNoResults()}
+
+        {data.map((item, index) => renderItem(item, index))}
+
+        {isFetching && renderSkeleton && renderSkeleton(pageSize)}
+
+        <div ref={loadMoreSentinelRef} style={{ height: '1px' }} />
+
+        {!hasMore && data.length > 0 && renderEndMessage()}
+      </div>
+    </div>
+  )
+}
+
+type Channel = Database['public']['Tables']['channels']['Row']
+
+// Define how each item should be rendered
+const renderChannelItem = (channel: Channel) => {
+  return (
+    <div
+      key={channel.id}
+      className="border-b py-3 px-4 hover:bg-muted flex items-center justify-between"
+    >
+      <div>
+        <span className="font-medium text-sm text-foreground">
+          {channel.slug} (ID: {channel.id})
+        </span>
+        <div className="text-sm text-foreground-light">
+          {new Date(channel.inserted_at).toLocaleDateString()}
+        </div>
+      </div>
+    </div>
+  )
+}
+
+// Define a filter to only show logs with log_level = 'info'
+const orderByInsertedAt: SupabaseQueryHandler<'channels'> = (query) => {
+  return query.order('inserted_at', { ascending: false })
+}
+
+const InfiniteListDemo = () => {
+  return (
+    <div className="bg-surface-100 h-[600px]">
```

Comment:
maybe wrong setup, but I'm getting all results at once 

![screenshot-2025-04-14-at-16 55 09](https://github.com/user-attachments/assets/094870a5-e126-40e2-899f-19fc2fac49cf)

Reply:
Addressed in the previous comment.